From 06b0e66547c063bc2c178bdd656834c603f77b0f Mon Sep 17 00:00:00 2001
From: Rob Browning <rlb@defaultvalue.org>
Date: Sun, 7 Apr 2013 15:03:05 -0500
Subject: Prefer /usr/share/info/emacs-MAJORVERSION

Emacs prefers /usr/share/info/emacs-MAJORVERSION to /usr/share/info.

The value of Info-default-directory-list has been augmented via
lisp/info.el to include /usr/share/info/emacs-MAJORVERSION before
/usr/share/info.
---
 lisp/info.el | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lisp/info.el b/lisp/info.el
index 6426cfcf9ed..d8a7f72e5dc 100644
--- a/lisp/info.el
+++ b/lisp/info.el
@@ -218,7 +218,8 @@ A header-line does not scroll with the rest of the buffer."
 	      (nconc standard-info-dirs (list config-dir))
 	    (cons config-dir standard-info-dirs))))
     (if (not (eq system-type 'windows-nt))
-	dirs
+        ;; Debian: add per-flavor info directory in front.
+        (cons (format "/usr/share/info/emacs-%d" emacs-major-version) dirs)
       ;; Include the info directory near where Emacs executable was installed.
       (let* ((instdir (file-name-directory invocation-directory))
 	     (dir1 (expand-file-name "../info/" instdir))
